### PR TITLE
Fixes #2091 path keywords extensions

### LINKED ISSUE #2091: Path keywords extensions: next/previous lumen segment
The idea is similar to #2086:
* introduce the new keywords LUMEN_NEXT_SEGMENT and LUMEN_PREVIOUS_SEGMENT (or LUMEN_PREV_SEGMENT ?)
* move/copy the list of the lumen segments from PK-Sim to the Core
https://github.com/Open-Systems-Pharmacology/PK-Sim/blob/e0f7adb278e0f80212114d00db29920d6eca24b8/src/PKSim.Core/CoreConstants.cs#L381-L382
* path resolution: similar to the LUMEN_SEGMENT keyword with the only difference, that **not the same** lumen segment is taken but the **next/previous** one, e.g.

- `Organism|Lumen|Stomach|LUMEN_NEXT_SEGMENT|P1` evaluates to `Organism|Lumen|Duodenum|P1`
- `Organism|Lumen|Stomach|LUMEN_PREV_SEGMENT|P1` - Error (no previous segment)
- `Organism|Lumen|Duodenum|LUMEN_PREV_SEGMENT|P1` evaluates to `Organism|Lumen|Stomach|P1`

This would allow simplifying e.g. the formulas below

grafik

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
@Yuri05 I think I have it :)

File: src/OSPSuite.Assets/UIConstants.cs [hunk 1]
```diff
@@ -1767,6 +1767,13 @@ public static string TimeArrayValuesDoesNotMatchFirstIndividual(int id, int inde
 
       public static string KeywordCanOnlyBeUsedOnce(string keyword, string path) => $"Keyword '{keyword}' can only be used once in '{path}'";
 
+      public static string LumenNavigationKeywordLCanOnlyBeUsedOnce(string path) => $"Only one lumen navigation keyword can be used once in '{path}'";
```
Comment (msevestre) [inline src/OSPSuite.Assets/UIConstants.cs:1770]: trying to be as explicit as possible for modeler here

File: src/OSPSuite.Core/Domain/Constants.cs [hunk 2]
```diff
@@ -128,6 +127,53 @@ public static class Files
...
+         public const string UPPER_JEJUNUM = "UpperJejunum";
+         public const string LOWER_JEJUNUM = "LowerJejunum";
+         public const string UPPER_ILEUM = "UpperIleum";
+         public const string LOWER_ILEUM = "LowerIleum";
+         public const string CAECUM = "Caecum";
+         public const string COLON_ASCENDENS = "ColonAscendens";
+         public const string COLON_TRANSVERSUM = "ColonTransversum";
+         public const string COLON_DESCENDENS = "ColonDescendens";
+         public const string COLON_SIGMOID = "ColonSigmoid";
+         public const string RECTUM = "Rectum";
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Constants.cs:148]: copied from PKSim

File: src/OSPSuite.Core/Domain/Formulas/FormulaExtensions.cs [hunk 3]
```diff
@@ -25,17 +25,7 @@ public static class FormulaExtensions
       /// </summary>
       public static bool IsExplicit(this IFormula formula) => formula.IsAnImplementationOf<ExplicitFormula>();
 
-      /// <summary>
-      ///    Returns true if the formula  contains at least one entry using neighborhood reference  otherwise false
-      /// </summary>
-      /// <param name="formula"></param>
-      public static bool IsReferencingNeighborhood(this IFormula formula) => formula.ObjectPaths.Any(x => x.Contains(ObjectPathKeywords.NBH));
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Formulas/FormulaExtensions.cs:32]: did not make sense to have those extension methods here

File: src/OSPSuite.Core/Domain/Services/FormulaTask.cs [hunk 4]
```diff
@@ -225,12 +243,53 @@ private void updateLumenSegmentReferencingPath(FormulaUsablePath formulaUsablePa
...
          var container = getContainerOrThrow(pathToContainer, usingFormula);
          //Point to our absolute ORGANISM|LUMEN|container path
-         var lumenSegmentPath = new List<string> {ORGANISM, LUMEN, container.Name};
+         var lumenSegmentPath = new List<string> {ORGANISM, Organs.LUMEN, container.Name};
          //we add the rest of the path that was provided after the keyword
          lumenSegmentPath.AddRange(pathAsList.Skip(lastIndex + 1));
          formulaUsablePath.ReplaceWith(lumenSegmentPath);
       }
 
+      private void updateLumenNavigationSegmentReferencingPath(FormulaUsablePath formulaUsablePath, IUsingFormula usingFormula, List<string> allLumenSegments)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Services/FormulaTask.cs:252]: this is the method here

**Review by msevestre — COMMENTED** [review]
Check it out and let me know :)

FYI @PavelBal new keywords

**Review by Yuri05 — APPROVED** [review]
looks good :)

**Pull request merged**